### PR TITLE
Fixes vector-im/element-ios/issues/5120 - Fixed fallback key signature validation.

### PR DESCRIPTION
**Pull request by stefanceriu**
Storing full MXKey dictionary so we can use all fields when checking the signature.
Added integration test to exercise currently failing paths.

### CODE REVIEW

File: MatrixSDK/Crypto/Data/MXKey.m [hunk 1]
```diff
@@ -85,22 +101,24 @@ + (id)modelFromJSON:(NSDictionary *)JSONDictionary
...
     }
     return JSONDictionary;
 }
 
 - (NSDictionary *)signalableJSONDictionary
 {
-    return @{
-             @"key": _value
-             };
+    if (_signalableJSONDictionary == nil) {
```
Comment (manuroe) [inline MatrixSDK/Crypto/Data/MXKey.m:117]:
```suggestion
    if (_signalableJSONDictionary == nil)
    {
```

File: MatrixSDKTests/MXCryptoTests.m [hunk 2]
```diff
@@ -3007,6 +3009,52 @@ - (void)testDiscardAndRestoreOlmOutboundKey
     }];
 }
 
+#pragma mark - One time / fallback keys
+
+- (void)testFallbackKeySignatures
+{
+    [MXSDKOptions sharedInstance].enableCryptoWhenStartingMXSession = YES;
```
Comment (manuroe) [inline MatrixSDKTests/MXCryptoTests.m:3016]: We need to reset it to `NO` after the MXSession creation to avoid to affect other tests. Yeah, singleton...

**Review by manuroe — COMMENTED** [review]
The code looks good to me.
@BillCarsonFr, can you double check that keys are aligned now?

File: MatrixSDKTests/MXCryptoTests.m (hunk 2 above)
Comment (stefanceriu) [reply MatrixSDKTests/MXCryptoTests.m:3016]: Ha, good one 😅

**Review by manuroe — APPROVED** [review]

**Review by BillCarsonFr — APPROVED** [review]

**Pull request merged**